scripts/validate_docstrings.py

```diff
@@ -491,8 +491,34 @@ def validate_one(func_name):
             errs.append('\t{}'.format(param_err))
 
     if doc.is_function_or_method:
-        if not doc.returns and "return" in doc.method_source:
-            errs.append('No Returns section found')
+        if not doc.returns:
+            if "return" in doc.method_source:
+                errs.append('No Returns section found')
```

Comment:
@datapythonista @WillAyd : How does this validation handle situations where we use an empty `return` to break out early from a function?  In such a case, we have a `return` statement, but the documentation wouldn't need to specifies a `Returns` section *per se*.

Reply:
Afaik, in Python all the functions return `None` by default, and a bare `return` also returns `None`. I think in those cases the numpydoc standard and ours is to not include the `Returns` section at all.

I don't think the validation checks in the code whether there is a return or not. This change should be to avoid giving a name to what is being returned if it's just a single value. In my opinion, saying that the function `transform` returns a variable `transformed` does not add any value.

We want to name the values returned if the return is a tuple, for example `code` and `name`. This PR should validate this.

Does this make sense to you @gfyoung?

Reply:
so replacing
`if "return" in doc.method_source:`
by something like
`if re.search(r"return\b[ \t]*[^;#\n\r]", doc.method_source)`
would deal with these cases ?

(maybe I should have another look at what's in doc.method_source)

Reply:
`doc.method_source` should be the code as a string. Not sure about the exact regex, not sure if looking for a return followed by letter, number or opening brackets would be simpler.

Reply:
Do we really need a regex for this? Wouldn't it just be better documented as an optional return value if such a branch is possible to hit?

Reply:
Not sure what you mean, but it's better if you mention this approach in #23488, where this will be implemented.